core/server/api/v0.1/decorators/urls.js

```diff
@@ -0,0 +1,86 @@
+const urlService = require('../../../services/url');
+const {urlFor, makeAbsoluteUrls} = require('../../../services/url/utils');
+
+const urlsForPost = (post, options) => {
+    post.url = urlService.getUrlByResourceId(post.id);
+
+    if (options.columns && !options.columns.includes('url')) {
+        delete post.url;
+    }
+
+    if (options && options.context && options.context.public && options.absolute_urls) {
+        if (post.feature_image) {
+            post.feature_image = urlFor('image', {image: post.feature_image}, true);
+        }
+
+        if (post.og_image) {
+            post.og_image = urlFor('image', {image: post.og_image}, true);
+        }
+
+        if (post.twitter_image) {
+            post.twitter_image = urlFor('image', {image: post.twitter_image}, true);
+        }
+
+        if (post.html) {
+            post.html = makeAbsoluteUrls(post.html, urlFor('home', true), post.url).html();
+        }
+
+        if (post.url) {
+            post.url = urlFor({relativeUrl: post.url}, true);
+        }
+    }
+
+    if (options && options.withRelated) {
```

Comment:
@kirrg001 was thinking if similar logic should be present in tag/user URL generation? Looks like includes similar to this are not allowed on those models (for example, only `count.posts` is allowed for tags)

Reply:
How did this work before?

Reply:
As discussed. We agree having hardcoded relations in here is not the nicest way to solve this problem, but in a given situation, it's the only option to make it work. Will be revisited once v2 is introduced.